### PR TITLE
Migration to AsciiDoc

### PR DESCRIPTION
**Pull request by max-rosin**
This PR is part of the migration to AsciiDoc.

### CODE REVIEW

**Comment by stefancyliax** [comment]
CCB 07.07.2021 addendum:
- Please review @pmai

File: doc/spec/sensor_data_inputs.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+= Sensor data inputs
+
+Sensor-data inputs are present in logical models.
+The sensor data passed to the model depends on prior models or processes that generated the data.
+The exact details of the contents therefore depend on the processing pipeline.
+
+The logical model can rely on the provided buffer remaining valid from the moment it is passed in until the end of the corresponding calculation.
+Itdoes not need to copy the contents (zero-copy input).
```
Comment (stefancyliax) [inline doc/spec/sensor_data_inputs.adoc]:
```suggestion
It does not need to copy the contents (zero-copy input).
```

File: doc/spec/sensor_data_outputs.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+= Sensor data outputs
+
+Sensor-data outputs are present in sensor models and logical models.
+
+The simulation engine can rely on the provided buffer to remain valid from the moment it is passed out until the end of the next co-simulation calculation cycle.
+Thus, the simulation engine does not need to copy the contents in that case, corresponding to zero-copy output for the simulation engine at the cost of double buffering for the sensor model.
+This arrangement is required to support use of the sensor model FMUs in simulation engines that have no special support for the protocol buffer pointers.
+Using this convention it is possible to daisy chain FMUs with protocol buffer inputs and outputs in a normal simulation engine, for example, MATLAB/Simulink, and get valid results.
```
Comment (stefancyliax) [inline doc/spec/sensor_data_outputs.adoc]:
```suggestion
Using this convention, it is possible to daisy chain FMUs with protocol buffer inputs and outputs in a normal simulation engine, for example, MATLAB/Simulink, and get valid results.
```

File: doc/spec/sensor_view_outputs.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
+= Sensor view outputs
+
+Sensor-view outputs are present in environmental effect models.
+
+The simulation engine can rely on the provided buffer to remain valid from the moment it is passed out until the end of the next co-simulation calculation cycle.
+Thus, the simulation engine does not need to copy the contents in that case, corresponding to zero-copy output for the simulation engine at the cost of double buffering for the environmental effect model.
+This arrangement is required to support use of the environmental effect model FMUs in simulation engines that have no special support for the protocol buffer pointers.
+Using this convention it is possible to daisy chain FMUs with protocol buffer inputs and outputs in a normal simulation engine, for example, MATLAB/Simulink, and get valid results.
```
Comment (stefancyliax) [inline doc/spec/sensor_view_outputs.adoc]:
```suggestion
Using this convention, it is possible to daisy chain FMUs with protocol buffer inputs and outputs in a normal simulation engine, for example, MATLAB/Simulink, and get valid results.
```

File: doc/spec/traffic_update_outputs.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
+= Traffic update outputs
+
+Traffic-update outputs are present in traffic participant models.
+
+The simulation engine can rely on the provided buffer to remain valid from the moment it is passed out until the end of the next co-simulation calculation cycle.
+Thus, the simulation engine does not need to copy the contents in that case, corresponding to zero-copy output for the simulation engine at the cost of double buffering for the traffic participant model.
+This arrangement is required to support use of the traffic participant model FMUs in simulation engines that have no special support for the protocol buffer pointers.
+Using this convention it is possible to daisy chain FMUs with protocol buffer inputs and outputs in a normal simulation engine, for example, MATLAB/Simulink, and get valid results.
```
Comment (stefancyliax) [inline doc/spec/traffic_update_outputs.adoc]:
```suggestion
Using this convention, it is possible to daisy chain FMUs with protocol buffer inputs and outputs in a normal simulation engine, for example, MATLAB/Simulink, and get valid results.
```

File: doc/spec/model_types.adoc [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+Traffic participant models::
+Consumes `osi3::SensorView` as input and produces `osi3::TrafficUpdate` as output.
+Traffic participant models may internally use environmental effect models, sensor models, or logical models as part of a modeled autonomous vehicle.
+They may also be used to implement surrounding traffic in simplified ways.
+Traffic participant models may consume `osi3::TrafficCommand` as input to allow control by a scenario engine as part of the simulation.
+
+All models may also consume a global `osi3::GroundTruth` during initialization.
+
+Complex models may combine various aspects of the above model types.
+However configuration and setup of such FMUs will require manual intervention.
```
Comment (stefancyliax) [inline doc/spec/model_types.adoc]:
```suggestion
However, configuration and setup of such FMUs will require manual intervention.
```

**Review by stefancyliax — CHANGES_REQUESTED** [review]
Found some small typos.

File: doc/spec/sensor_data_inputs.adoc (hunk 1 above)
Comment (max-rosin) [reply doc/spec/sensor_data_inputs.adoc]: Changed.

File: doc/spec/sensor_data_outputs.adoc (hunk 2 above)
Comment (max-rosin) [reply doc/spec/sensor_data_outputs.adoc]: Changed.

File: doc/spec/sensor_view_outputs.adoc (hunk 3 above)
Comment (max-rosin) [reply doc/spec/sensor_view_outputs.adoc]: Changed.

File: doc/spec/traffic_update_outputs.adoc (hunk 4 above)
Comment (max-rosin) [reply doc/spec/traffic_update_outputs.adoc]: Changed.

File: doc/spec/model_types.adoc (hunk 5 above)
Comment (max-rosin) [reply doc/spec/model_types.adoc]: Changed.

**Review by stefancyliax — DISMISSED** [review]
All feedback was included.

**Review by pmai — APPROVED** [review]
With the given rewrite I think this now transports the same normative details as the previous version while trying to fit better into the new mold of documentation.

**Comment by pmai** [comment]
@stefancyliax can you review the new changes, if this fits the immediate needs we could merge and release (NB: for release version updates, submodule updates, etc. need to be performed). This still leaves some of the other improvements in the pending PRs out of the release though...

**Review by stefancyliax — APPROVED** [review]
Reviewed rework. Looks fine.

**Comment by pmai** [comment]
> Reviewed rework. Looks fine.

Can you check whether this also works in terms of the overall documentation pipeline? I have not kept track with that...

**Comment by stefancyliax** [comment]
> > Reviewed rework. Looks fine.
> 
> Can you check whether this also works in terms of the overall documentation pipeline? I have not kept track with that...

Already on it. OSMP document itself builds fine. Pipeline for osi-documentation also ok.

**Pull request merged**